### PR TITLE
Move tickets with recent replies to the top

### PR DESCRIPTION
**Pull request by AGSoftAU**
See #271

### CODE REVIEW

**Comment by SiamKreative** [comment]
What would be the performance impact?

**Comment by AGSoftAU** [comment]
The performance impact of the changes made are marginal at best. The changes made are only present in the dashboard, and the actual query is edited when it's needed.

**Comment by julien731** [comment]
Hey Alex. Sorry for not replying earlier. Thanks a lot for this PR that looks really cool. I haven't had time to work on AS lately but as soon as I get back to it your PR will be merge. Many thanks for the work.

**Comment by julien731** [comment]
It's been a while now but we haven't forgotten about this PR! We were planning on merging it into version 3.3 which we are working on at the moment, but unfortunately, this won't work.

The base idea of these changes is good. However, it won't achieve the desired result. What we want to do is have *tickets that need to be replied to by an agent* at the top.

What this PR does is place the tickets with the most recent replies at the top. The agent factor is taken off of the equation here.

With these changes, tickets recently replied to will be shown at the top, **even if the last answer was from an agent**.

## Example

Let's imagine that I have two tickets. Ticket A, opened in the morning, and ticket B, opened in the afternoon. Both tickets are new and unanswered.

Because of their opening dates, ticket A will appear before ticket B in the admin.

Agent Joe starts his shift and replies to ticket A. This leaves us with ticket A answered, and ticket B unanswered.

What we want to achieve is to have ticket B appear first, and then ticket A, because ticket B is the one awaiting reply now.

However, because this PR sorts tickets by the most recent date of reply, ticket A will appear first because the most recent reply is the one made to ticket A **by agent John**.

In conclusion, the base query is good, but we need to somehow factor in the capability of the user who replied last in order to differentiate agents and clients.

**Comment by julien731** [comment]
Good start but needs more work.
